plugins/MantisCoreFormatting/core/MantisMarkdown.php

```diff
@@ -42,220 +42,212 @@
+		return $this->codeblocks;
 	}
 
 	/**
-	 * Customize the inlineCode method
+	 * Convert an array of element data into the HTML markup.
 	 *
-	 * @param array $block A block-level element
-	 * @access protected
-	 * @return string html representation generated from markdown.
+	 * @param array $Element Data for an element
+	 * @return string HTML markup for an element
 	 */
-	protected function inlineCode( $block ) {
-
-		$block = parent::inlineCode( $block );
+	protected function element( array $Element ): string {
+		# Capture the code blocks to prevent them from being processed further.
+		if( $Element['name'] === 'code' ) {
+			$t_hash = $this->hash( $Element['text'] );
+			$this->codeblocks[$t_hash] = $Element['text'];
+			$Element['text'] = $t_hash;
+		}
 
-		if( isset( $block['element']['text'] )) {
-			$this->processAmpersand( $block['element']['text'] );
+		# Adding CSS classes to tables.
+		if( $Element['name'] === 'table' ) {
+			$Element['attributes']['class'] = $this->table_class;
 		}
 
-		return $block;
+		return parent::element( $Element );
 	}
 
 	/**
-	 * Customize the blockFencedCodeComplete method
+	 * Convert an email addresses in unmarked text into a link.
 	 *
-	 * @param array $block A block-level element
-	 * @access protected
-	 * @return string html representation generated from markdown.
+	 * Unlike unmarked URLs, unmarked email addresses are not
+	 * processed by Parsedown.
 	 */
-	protected function blockFencedCodeComplete( $block = null ) {
-
-		$block = parent::blockFencedCodeComplete( $block );
-
-		if( isset( $block['element']['text']['text'] )) {
-			$this->processAmpersand( $block['element']['text']['text'] );
+	protected function unmarkedText( $text ): string {
+		if( ON == $this->config_process_urls && false !== strpos( $text, '@' ) ) {
+			$text = string_insert_hrefs( $text );
 		}
 
-		return $block;
+		return parent::unmarkedText( $text );
 	}
 
 	/**
-	 * Customize the blockCodeComplete method
+	 * Implementation of the CommonMark specification for headers,
+	 * as Parsedown does not follow the specifications.
 	 *
-	 * @param array $block A block-level element
-	 * @access protected
-	 * @return string html representation generated from markdown.
+	 * @see https://spec.commonmark.org/0.31.2/#example-62
+	 * @see https://parsedown.org/demo
+	 *
+	 * @param array $Line Data for the block element
+	 * @return array|void Only return data for the header element if it matches
+	 *                    the specification
 	 */
-	protected function blockCodeComplete( $block ) {
-
-		$block = parent::blockCodeComplete( $block );
-
-		if( isset( $block['element']['text']['text'] )) {
-			$this->processAmpersand( $block['element']['text']['text'] );
+	protected function blockHeader( $Line ) {
+		# - the opening # may be indented 0-3 spaces
+		# - a sequence of 1–6 '#' characters
+		# - The #'s must be followed by a space or a newline
+		if ( preg_match( '/^ {0,3}#{1,6}(?: |$)/', $Line['text'] ) ) {
```

Comment:
Commonmark says *The opening sequence of # characters must be followed by spaces or tabs, or by the end of line.* So this needs a little adjustment to be compliant:

```php
# - The #'s must be followed by a space, a tab or a newline
if ( preg_match( '/^ {0,3}#{1,6}(?:[ \t]|$)/', $Line['text'] ) ) {
```